packages/notebook/src/widget.ts

```diff
@@ -201,6 +202,11 @@ export class StaticNotebook extends Widget {
    */
   readonly rendermime: RenderMimeRegistry;
 
+  /**
+   * The base URL of the application, used to compute nbconvert URL.
+   */
+  readonly baseUrl?: string;
```

Comment:
I don't think the notebook widget should have to know anything about the `baseUrl`, since we want it to be able to be rendered statically. I think for the purposes of printing to `nbconvert`, we can get that information elsewhere.

Reply:
I had some trouble figuring out how to thread this in properly. The underlying issue is that the print method is [defined on the panel](https://github.com/jupyterlab/jupyterlab/pull/5850/files/92880b15b525688f22b9fd186b6e1a4fe7a45244#diff-30a0e8e264c5bcc2eeae28f7c78829b8R183), therefore the panel needs to know the base URL so it can calculate the nbconvert URL.

I will take another look through this so I can see a way around it...

Reply:
I was able to remove this, since `baseUrl` is known globally in the `PageConfig`.